shared/core/src/main/scala/org/hyperledger/identus/shared/messaging/kafka/ZKafkaMessagingServiceImpl.scala

```diff
@@ -0,0 +1,118 @@
+package org.hyperledger.identus.shared.messaging.kafka
+
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.common.header.Headers
+import org.hyperledger.identus.shared.messaging.*
+import zio.{durationInt, EnvironmentTag, RIO, RLayer, Task, ULayer, URIO, ZIO, ZLayer}
+import zio.kafka.consumer.{
+  Consumer as ZKConsumer,
+  ConsumerSettings as ZKConsumerSettings,
+  Subscription as ZKSubscription
+}
+import zio.kafka.producer.{Producer as ZKProducer, ProducerSettings as ZKProducerSettings}
+import zio.kafka.serde.{Deserializer as ZKDeserializer, Serializer as ZKSerializer}
+
+class ZKafkaMessagingServiceImpl(bootstrapServers: List[String]) extends MessagingService {
+  override def makeConsumer[K, V](groupId: String)(implicit kSerde: Serde[K], vSerde: Serde[V]): Task[Consumer[K, V]] =
+    ZIO.succeed(new ZKafkaConsumerImpl[K, V](bootstrapServers, groupId, kSerde, vSerde))
+
+  override def makeProducer[K, V]()(implicit kSerde: Serde[K], vSerde: Serde[V]): Task[Producer[K, V]] =
+    ZIO.succeed(new ZKafkaProducerImpl[K, V](bootstrapServers, kSerde, vSerde))
+}
+
+object ZKafkaMessagingServiceImpl {
+  def layer(bootstrapServers: List[String]): ULayer[MessagingService] =
+    ZLayer.succeed(new ZKafkaMessagingServiceImpl(bootstrapServers))
+}
+
+class ZKafkaConsumerImpl[K, V](
+    bootstrapServers: List[String],
+    groupId: String,
+    kSerde: Serde[K],
+    vSerde: Serde[V]
+) extends Consumer[K, V] {
+  private val zkConsumer = ZLayer.scoped(
+    ZKConsumer.make(
+      ZKConsumerSettings(bootstrapServers)
+        .withProperty(ConsumerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG, false.toString)
```

Comment:
Should we move all these configuration parameters to the config file?

Reply:
Yes, this has been implemented recently 👍